extensions/message-ids.md

```diff
@@ -0,0 +1,130 @@
+---
+title: IRCv3 Message IDs
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "James Wheare"
+    period: 2016
+    email: "james@irccloud.com"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `msgid` tag name. Instead, implementations SHOULD use
+the `draft/msgid` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Introduction
+
+This specification describes a mesage tag indicating a server supplied unique ID for `PRIVMSG` and `NOTICE` messages.
```

Comment:
Any reason this doesn't apply to other events as well such as all channel events, NICK and QUIT? Having a unique id on those events as well allows clients to deduplicate all messages in case of e.g. history playback.